### DISCUSSION TITLE
archetype wizard runs in interactive mode even with defaults for required properties

### DISCUSSION THREAD

**Discussion by krabat** [eclipse-m2e/m2e-core#1313] [Q&A]
When creating a new maven project, based on my archetype, I am only allowed to run it in "interactive" mode. There (eclipse console),  I have to provide values for my "requiredProperties", although I defined defaults for them, which I don't want to override. I want the project creation run in non-interactive mode.

The last time I ran it (many month ago), it used non-interactive mode automatically and just generated the project with my defaults. But back then, I noticed my properties (project-name, project-description) were listed in the "Properties available from archetype", which is not the case anymore. 

My archetype-metadata.xml:
```
	<requiredProperties>
		<requiredProperty key="project-name">
			<defaultValue>${artifactId}</defaultValue>
		</requiredProperty>
		<requiredProperty key="project-description">
			<defaultValue>${artifactId} description</defaultValue>
		</requiredProperty>
		<!-- The launcher RUN requires the path to the main class file, including the slashed package -->
		<requiredProperty key="package-with-slash">
			<defaultValue>${package.replace(".", "/")}</defaultValue>
		</requiredProperty>
	</requiredProperties>
```

**Comment by laeubi** [comment]
Please provide a runnable example if you think there is a problem, beside this, interactive mode can be disabled here:
![grafik](https://user-images.githubusercontent.com/1331477/224723685-c0917606-d7a8-4af4-bc56-fa8d5486a476.png)

**Comment by krabat** [comment]
Sorry, I should have included the fact, that I indeed unchecked the interactive mode, but it won't let my create my project:

![m2e-archetype](https://user-images.githubusercontent.com/971266/224725277-e78b5a60-0894-495f-8ea0-546162b15640.jpg)

**Reply by laeubi** [reply]
The error message tells you that interactive mode is required this is because your archetype is using variable in the default values that can only be evaluated at runtime.

**Reply by krabat** [reply]
Ok, but the wizard's behavior was different back then: before I was able to let the wizard run automatically (non-interactive).

Any suggestion/workaround on how I can achieve the former behavior again with my current required properties setup?

**Reply by laeubi** [reply]
No this is not possible, but you should be able to simply accept the default values, that is just standard maven behavior.

**Reply by krabat** [reply]
It's sad the user experience got worse and the usability suffers from it. Personally I would like the former behavior back (e.g. via 'force non-interactive' or a warning).

But I want to thank you @laeubi  for answering and assisting me so quickly.

**Reply by laeubi** [reply]
Its all depends a bit previously there where no validation and no support for groovy scripts or never archetype features. So that it previously worked for this case don't mean it was correct. If you want full non-interactive I think it works when you specify (with add) a value but then obviously the defaults can not work (unless someone implements this what is a non trivial task).
